search/main.py

```diff
@@ -57,60 +56,47 @@ def __init__(self):
         }
 
         self.distinct_attribute = "url"
-
-    def get_host(self):
+    
+    def getConnectionString(self):
         """
         Sets the Meilisearch host string, depending on the environment.
 
         Returns:
             string: Meilisearch host string.
         """
-        port = 7700
-        if self.env() == 'prod' or os.environ.get('PORT'):
-            port = os.environ['PORT']
-        return "{}:{}".format(self.localhost, port)
+        if os.environ.get('PORT'):
+            return "{}:{}".format(self.default["host"], os.environ['PORT'])
+        else:
+            return "{}:{}".format(self.default["host"], self.default["port"])
 
-    def env(self):
+    def getMasterKey(self):
         """
-        Sets the Meilisearch environment, depending on the presence of a master key.
-
-        Returns:
-            string: 'prod' if MEILI_MASTER_KEY is set, 'dev' otherwise.
+        Retrieves the Meilisearch master key, either from the Platform.sh environment or locally.
         """
-        if os.environ.get('MEILI_MASTER_KEY')==None:
-            return 'dev'
-        elif os.environ.get('PORT')==None:
-            os.environ['PORT'] = self.default_port
-            return 'prod'
+        config = Config()
+        if config.is_valid_platform():
+            return config.projectEntropy
+        elif os.environ.get("MEILI_MASTER_KEY"):
+            return os.environ["MEILI_MASTER_KEY"]
         else:
-            return 'prod'
+            return self.default["key"]
+    
+    def add_documents(self, index):
+        documents = [f for f in glob.glob("{}/*.json".format(self.scrape_dir))]
+        for docs in documents:
+            self.add(docs, index)
 
+    def add(self, docs, index):
+        with open(docs) as scraped_index:
+            data = json.load(scraped_index)
+            index.add_documents(data)
```

Comment:
```suggestion
    def add_documents(self, index):
        documents = [f for f in glob.glob("{}/*.json".format(self.scrape_dir))]
        for doc in documents:
            self.add(doc, index)

    def add(self, doc, index):
        with open(doc) as scraped_index:
            data = json.load(scraped_index)
            index.add_documents(data)
```

This confused me for a while, as it seemed you were reopening a group of files over and over again.

Reply:
Is this in any way parallelizable?